tests/integration/targets/cloud_discovery_providers_aws/tasks/main.yml

```
@@ -0,0 +1,485 @@
+      register: aws_provider
+    - name: Get AWS cloud discovery provider
+      infoblox.universal_ddi.cloud_discovery_providers_info:
+        filters:
+          name: "{{ cloud_discovery_provider_aws_name }}"
+      register: aws_provider_info
+    - assert:
+        that:
+          - aws_provider is changed
+          - aws_provider_info is not failed
+          - aws_provider_info.objects | length == 0
+
+    - name: Create an AWS cloud discovery provider
+      infoblox.universal_ddi.cloud_discovery_providers:
+        name: "{{ cloud_discovery_provider_aws_name }}"
+        provider_type: "Amazon Web Services"
+        account_preference: "single"
+        credential_preference:
+          access_identifier_type: "role_arn"
+          credential_type: "dynamic"
+        source_configs:
+          - credential_config:
+                access_identifier: "{{ config_access_id }}"
+        state: present
+      register: aws_provider
+    - name: Get AWS cloud discovery provider
+      infoblox.universal_ddi.cloud_discovery_providers_info:
+        filters:
+          name: "{{ cloud_discovery_provider_aws_name }}"
+      register: aws_provider_info
+    - assert:
+        that:
+          - aws_provider is changed
+          - aws_provider_info is not failed
+          - aws_provider_info.objects | length == 1
+          - aws_provider_info.objects[0].name == cloud_discovery_provider_aws_name
+          - aws_provider_info.objects[0].provider_type == "Amazon Web Services"
+          - aws_provider_info.objects[0].account_preference == "single"
+          - aws_provider_info.objects[0].credential_preference.access_identifier_type == "role_arn"
+          - aws_provider_info.objects[0].credential_preference.credential_type == "dynamic"
+
+    - name: Create an AWS cloud discovery provider (idempotent)
+      infoblox.universal_ddi.cloud_discovery_providers:
+        name: "{{ cloud_discovery_provider_aws_name }}"
+        provider_type: "Amazon Web Services"
+        account_preference: "single"
+        credential_preference:
+          access_identifier_type: "role_arn"
+          credential_type: "dynamic"
+        source_configs:
+          - credential_config:
+                access_identifier: "{{ config_access_id }}"
+        state: present
+      register: aws_provider
+    - assert:
+        that:
+          - aws_provider is not changed
+          - aws_provider is not failed
+
+    - name: Delete an AWS cloud discovery provider (check mode)
+      infoblox.universal_ddi.cloud_discovery_providers:
+        name: "{{ cloud_discovery_provider_aws_name }}"
+        provider_type: "Amazon Web Services"
+        account_preference: "single"
+        credential_preference:
+          access_identifier_type: "role_arn"
+          credential_type: "dynamic"
+        source_configs:
+          - credential_config:
+                access_identifier: "{{ config_access_id }}"
+        state: absent
+      check_mode: yes
```

```suggestion
      check_mode: true
```
Please change all the occurrences of check mode to true